core/io/resource.h

```diff
@@ -104,7 +104,7 @@ class Resource : public RefCounted {
 	virtual void set_path(const String &p_path, bool p_take_over = false);
 	String get_path() const;
 	void set_path_cache(const String &p_path); // Set raw path without involving resource cache.
-	_FORCE_INLINE_ bool is_built_in() const { return path_cache.is_empty() || path_cache.contains("::") || path_cache.begins_with("local://"); }
+	_FORCE_INLINE_ bool is_built_in(String p_local_path = "") const { return path_cache.is_empty() || (path_cache.contains("::") && (path_cache.begins_with(p_local_path) || p_local_path == "")) || path_cache.begins_with("local://"); }
```

Comment:
Feels like it may be better to have a separate function for this, like `is_built_in_for_local_path`